bool is_inj
---
bool ok
